src/api/firebase.ts

```diff
@@ -301,3 +301,44 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(item1: ListItem, item2: ListItem) {
+	const currentDate = new Date();
+
+	// Check for last time item had any activity
+	const daysSinceItemLastActivity = item1.dateLastPurchased
+		? getDaysBetweenDates(currentDate, item1.dateLastPurchased.toDate())
+		: getDaysBetweenDates(currentDate, item1.dateCreated.toDate());
+
+	// Prioritize item1 if current date is past next purchase date
+	if (currentDate > item1.dateNextPurchased.toDate()) {
+		return -1;
```

Comment:
I really like this! Did you consider an enum here? There's pros and cons:
- Enums are much easier to read/understand (and therefore produce more maintainable* code). You could have an enum that contains three variants: `Priority.Low`, `Priority.Medium`, `Priority.High`. Now you don't need to write down what `-1` means. To some, that might be misunderstood to be _very_ low priority. Careful documentation can get around that. You can also assign enums values ([see some of the examples here](https://www.typescriptlang.org/docs/handbook/enums.html)) so you can still keep the numbers where you need them (if we're putting this in a database, for instance) but give them "more meaning" in the code.
- Numbers along are really great if you plan on adding more priorities in the future, which isn't _really_ a thing here, but something to keep in mind. This is something I've seen done with errors before, where a function will have a big switch statement against every error variant a function could return, and assign a numerical value to the significance of that error. Assuming you want to trust someone else's evaluation of "significance" there, I don't think I would in production code but i'm just paranoid, you could decide to handle the "top 30% of significant errors" differently than "lower priority" errors. Imo this is typically the result of less-than-great api design and could be avoided in other ways, but I'll try to avoid a soap box here 😄 

tldr; maybe consider enums with values assigned to them 🚀

Reply:
🤦  I just realized this is for a `sort` - forget everything I said! Maybe we could leave a comment above this function explaining what it's used for and a brief overview of what it considers a "high priority" vs "low priority" ?